Issue with "ab-suite.git.clone.stripped.log"

Running the script it give a duplicate error for "ab-suite.git.clone.stripped.log" after deleting it it continues to the next package/aplication git en then breaks again.

Example:

>        Starting 32bit compilation of all tools
>Imported options from ffmpeg_options.txt
>Imported custom options from ffmpeg_options.txt
>License: nonfree
>Imported options from mpv_options.txt
>
>        Starting 32bit compilation of global tools
>23:15:17 + Running rustup_update...
>23:15:17 + Running set_default_toolchain...
>23:15:18   Running git update for ripgrep...
>23:15:19 ripgrep git ...................................... [Up-to-date]
>23:15:20   Running git update for oniguruma...
>23:15:21 oniguruma git .................................... [Up-to-date]
>23:15:21   Running git update for jq...
>23:15:23 jq git ........................................... [Up-to-date]
>23:15:23   Running git update for dssim...
>23:15:25 dssim git ........................................ [Up-to-date]
>23:15:27   Running git clone for freetype2...
>23:20:15   Creating diagnostics file...
>
>
>ERROR:
>Duplicate filename on disk:
>ab-suite.git.clone.stripped.log
>ab-suite.git.clone.stripped.log
>
>Failed cloning to freetype2-git

after deleting the log:

>        Starting 32bit compilation of all tools
>Imported options from ffmpeg_options.txt
>Imported custom options from ffmpeg_options.txt
>License: nonfree
>Imported options from mpv_options.txt
>
>        Starting 32bit compilation of global tools
>23:29:47 + Running rustup_update...
>23:29:47 + Running set_default_toolchain...
>23:29:48   Running git update for ripgrep...
>23:29:49 ripgrep git ...................................... [Up-to-date]
>23:29:50   Running git update for oniguruma...
>23:29:51 oniguruma git .................................... [Up-to-date]
>23:29:51   Running git update for jq...
>23:29:53 jq git ........................................... [Up-to-date]
>23:29:53   Running git update for dssim...
>23:29:54 dssim git ........................................ [Up-to-date]
>23:29:58   Running git update for freetype2...
>23:30:00 freetype2 git .................................... [Up-to-date]
>23:30:00   Running git clone for fontconfig...
>23:30:05   Creating diagnostics file...
>
>
>ERROR:
>Duplicate filename on disk:
>ab-suite.git.clone.stripped.log
>ab-suite.git.clone.stripped.log
>
>Failed cloning to fontconfig-git

The issue is connected with building the freetype2 package.
After cloning the freetype2-git package it tries to run ./configure in build/freetype2-git/builds/unix directory, but there's no file named like that.

```
┌ freetype2 git ........................................ [Files missing]
├ Running configure...
Likely error:
* module: pshinter  (Postscript hinter module)
* module: raster    (monochrome bitmap renderer)
* module: smooth    (anti-aliased bitmap renderer)
* module: smooth    (anti-aliased bitmap renderer for LCDs)
* module: smooth    (anti-aliased bitmap renderer for vertical LCDs)
* module: psaux     (Postscript Type 1 & Type 2 helper module)
* module: psnames   (Postscript & Unicode Glyph name handling)
/build/freetype2-git/builds/unix/configure  '--build=x86_64-w64-mingw32' '--host=x86_64-w64-mingw32' '--target=x86_64-w64-mingw32' '--prefix=/local64' '--disable-shared' '--enable-static' '--bindir=/local64/bin-global' '--with-harfbuzz=no' '--with-png=no' '--with-bzip2=no'
make: /build/freetype2-git/builds/unix/configure: Command not found
make: *** [/build/freetype2-git/builds/unix/detect.mk:89: setup] Error 127
```

However, there is a "configure.raw" file. You can try change the name of the file to "configure", but it still outputs and error:

```
┌ freetype2 git ........................................ [Files missing]
├ Running configure...
Likely error:
* module: raster    (monochrome bitmap renderer)
* module: smooth    (anti-aliased bitmap renderer)
* module: smooth    (anti-aliased bitmap renderer for LCDs)
* module: smooth    (anti-aliased bitmap renderer for vertical LCDs)
* module: psaux     (Postscript Type 1 & Type 2 helper module)
* module: psnames   (Postscript & Unicode Glyph name handling)
/build/freetype2-git/builds/unix/configure  '--build=x86_64-w64-mingw32' '--host=x86_64-w64-mingw32' '--target=x86_64-w64-mingw32' '--prefix=/local64' '--disable-shared' '--enable-static' '--bindir=/local64/bin-global' '--with-harfbuzz=no' '--with-png=no' '--with-bzip2=no'
/build/freetype2-git/builds/unix/configure: line 14: syntax error near unexpected token `[FreeType],'
/build/freetype2-git/builds/unix/configure: line 14: `AC_INIT([FreeType], [@VERSION@], [freetype@nongnu.org], [freetype])'
make: *** [/build/freetype2-git/builds/unix/detect.mk:89: setup] Error 2
```

> 
> 
> Let me clarify something because I think there are 2 issues going on here:
> 
>     1. The system encounters an issue with the git / clone logic within `do_vcs()`
> 
>     2. The system goes to exit and log the diagnostic information about the error, `do_exit_prompt()` -> `zip_files()`, and we receive yet an additional error about the *****.log file already existing on disk.

Just like you wrote - there are two issues.

After the initial update, which ends with the exact same error as @GDXN  posted and then deleting the .log file, the autobuild is capable of creating a ziplog file: https://0x0.st/zT2u.zip

And there, in the compile.stripped.log:
```
  Running git update for freetype2...
┌ freetype2 git ........................................ [Files missing]
├ Running configure...
Likely error:
* module: pshinter  (Postscript hinter module)
* module: raster    (monochrome bitmap renderer)
* module: smooth    (anti-aliased bitmap renderer)
* module: smooth    (anti-aliased bitmap renderer for LCDs)
* module: smooth    (anti-aliased bitmap renderer for vertical LCDs)
* module: psaux     (Postscript Type 1 & Type 2 helper module)
* module: psnames   (Postscript & Unicode Glyph name handling)
/build/freetype2-git/builds/unix/configure  '--build=x86_64-w64-mingw32' '--host=x86_64-w64-mingw32' '--target=x86_64-w64-mingw32' '--prefix=/local64' '--disable-shared' '--enable-static' '--bindir=/local64/bin-global' '--with-harfbuzz=no' '--with-png=no' '--with-bzip2=no'
make: /build/freetype2-git/builds/unix/configure: Command not found
make: *** [/build/freetype2-git/builds/unix/detect.mk:89: setup] Error 127
configure failed. Check C:/Users/DJ/Desktop/media-autobuild_suite-master/build/freetype2-git/build-64bit/ab-suite.configure.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...
```

ab-suite.configure.log:

```
/build/freetype2-git/builds/unix/configure  '--build=x86_64-w64-mingw32' '--host=x86_64-w64-mingw32' '--target=x86_64-w64-mingw32' '--prefix=/local64' '--disable-shared' '--enable-static' '--bindir=/local64/bin-global' '--with-harfbuzz=no' '--with-png=no' '--with-bzip2=no'
make: /build/freetype2-git/builds/unix/configure: Command not found
make: *** [/build/freetype2-git/builds/unix/detect.mk:89: setup] Error 127
```

--- a/build/media-suite_helper.sh
+++ b/build/media-suite_helper.sh
@@ -226,8 +226,6 @@ do_vcs() {
             vcsURL="https://gitlab.com/media-autobuild_suite-dependencies/${repoName}"
             log quiet "$vcsType.clone" vcs_clone ||
             do_exit_prompt "Failed cloning to $vcsFolder-$vcsType"
-        else
-            do_exit_prompt "Failed cloning to $vcsFolder-$vcsType"
         fi
         if [[ -d "$vcsFolder-$vcsType" ]]; then
             cd_safe "$vcsFolder-$vcsType"

